### PR TITLE
chore: Use chanzuckerberg/github-actions/docker-build-push in Push Remote Dev action

### PR DESCRIPTION
**Pull request by edulop91**
### Summary:
- **What:** Starting to look into integrating the new happy cli with GitHub actions. As part of this, looking to release a set of common chanzuckerberg/github-actions. Chose the docker step first because it is fairly self-contained and common across many repos. Will follow up with Happy specific actions later.

See https://github.com/chanzuckerberg/github-actions/pull/17/ for corresponding code changes.

- **Ticket:** NA
- **Env:** `<rdev link>`

### Checklist:
- [ ] I merged latest `<base branch>`
- [ ] I manually verified the change
- [ ] I added labels to my PR
- [x] I tested in multiple browsers
- [x] I added relevant unit tests
- [x] I have notified others of changes they need to make locally (migrations, jobs, package updates, etc)

### CODE REVIEW

**Comment by edulop91** [comment]
hmm think i'm missing some build keys

File: .github/workflows/push-rdev.yml [hunk 1]
```diff
@@ -1,9 +1,9 @@
 name: Push Remote Dev
 
 on:
-  push:
+  pull_request:
```
Comment (jgadling) [inline .github/workflows/push-rdev.yml:4]: I think we specifically want to use the `push` event here

**Review by jgadling — CHANGES_REQUESTED** [review]
This looks good aside from the event hook!

**Comment by edulop91** [comment]
see https://github.com/chanzuckerberg/czgenepi/pull/990 instead for the push event instead of pull_request

**Pull request closed without merging**